internal/runbits/buildplanner/buildplanner.go

```diff
@@ -125,15 +126,25 @@ func GetCommit(
 		commitUUID = *branchCommitUUID
 
 		bp := bpModel.NewBuildPlannerModel(auth, svcm)
-		commit, err = bp.FetchCommit(commitUUID, namespace.Owner, namespace.Project, targetPtr)
+		commit, err = bp.FetchCommit(commitUUID, namespace.Owner, namespace.Project, branch.Label, targetPtr)
 		if err != nil {
 			return nil, errs.Wrap(err, "Failed to fetch commit")
 		}
 
 	// Return the buildplan for the given commitID of the given project.
 	case namespaceProvided && commitIdProvided:
+		pj, err := model.FetchProjectByName(namespace.Owner, namespace.Project, auth)
+		if err != nil {
+			return nil, locale.WrapExternalError(err, "err_fetch_project", "", namespace.String())
+		}
+
+		branch, err := model.DefaultBranchForProject(pj)
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not grab branch for project")
+		}
```

Comment:
We should provide this explicitly. Because we may not be on the default branch, and at this level we have on idea what the intend was. Consuming code won't know that this is setting up any branch related stuff.

Note we could just add an argument for the branch which can be nil, and then we default to this logic if it is nil. My concern is primarily about making the contract with the branch explicit with consuming code.

Reply:
We can actually just pass "", and the build script checkout info will silently ignore it (like activestate.yaml with just a commitID and no branch). We cannot explicitly provide a branch argument because `state export buildplan`, `state artifacts` and `state artifacts download` only know about commit ID. We'd have to do branch lookup or ask the user to supply it on the command line. Either one is messy.

Let me know if you disagree.